recipes/dns_config.rb

```diff
@@ -0,0 +1,148 @@
+    end
+  else
+    Chef::Log.info("Appending search domain '#{node['cfncluster']['cfn_dns_domain']}' to /etc/dhcp/dhclient.conf")
+    # Configure dhclient to automatically append Route53 search domain in resolv.conf
+    # - on CentOS6 resolv.conf is managed by network + dhclient,
+    # - on CentOS7, Alinux and Alinux2 by NetworkManager + dhclient,
+    # - on Ubuntu16 by networking + dhclient
+    replace_or_add "Append Route53 search domain in /etc/dhcp/dhclient.conf" do
+      path "/etc/dhcp/dhclient.conf"
+      pattern "append domain-name*"
+      line "append domain-name \" #{node['cfncluster']['cfn_dns_domain']}\";"
+    end
+
+    if platform?('centos') && node['platform_version'].to_i < 7
+      # On CentOS6 there is a dhclient configuration file for eth0
+      replace_or_add "Append Route53 search domain in /etc/dhcp/dhclient-eth0.conf" do
+        path "/etc/dhcp/dhclient-eth0.conf"
+        pattern "append domain-name*"
+        line "append domain-name \" #{node['cfncluster']['cfn_dns_domain']}\";"
+      end
+    end
+  end
+end
+
+# Utility method to restart network service according to the OS.
+def restart_network_service
+  network_service_name = value_for_platform(
+    ['centos'] => {
+      '<7.0' => 'network'
+    },
+    ['amazon'] => {
+      '>=2013' => 'network' # Alinux1
+    },
+    %w[ubuntu debian] => {
+      '16.04' => 'networking',
+      '>=18.04' => 'systemd-resolved'
+    },
+    'default' => 'NetworkManager'
+  )
+  Chef::Log.info("Restarting '#{network_service_name}' service, for platform #{node['platform']} '#{node['platform_version']}'")
+  service network_service_name.to_s do
+    action %i[restart]
+  end
+end
+
+if node['cfncluster']['cfn_scheduler'] == 'slurm'
+  # Heterogeneous Instance Type
+  if node['cfncluster']['cfn_dns_domain'].nil? ||
+     node['cfncluster']['cfn_dns_domain'].empty? ||
+     node['cfncluster']['cfn_dns_domain'] == "NONE"
+    Chef::Log.info("DNS domain not configured")
+    return
+  end
+
+  append_cluster_dns_domain
+
+  restart_network_service
+
+  if node['cfncluster']['cfn_node_type'] == "ComputeFleet"
+    # For compute node retrieve assigned hostname from DynamoDB and configure it
+    # - hostname: $QUEUE-static-$INSTANCE_TYPE_1-[1-$MIN1]
+    # - fqdn: $QUEUE-static-$INSTANCE_TYPE_1-[1-$MIN1].$CLUSTER_NAME.pcluster
+    Chef::Log.info("Retrieving assigned hostname")
+    begin
+      retries ||= 0
+      assigned_hostname = retrieve_assigned_hostname
+      raise if assigned_hostname == "None"
+
+      configure_hostname(assigned_hostname)
+    rescue StandardError
+      raise if (retries += 1) >= 5
+      Chef::Log.warn("Failed when retrieving assigned hostname, retrying in 3 seconds...")
+      sleep(3)
+      retry
+    end
+
+  else
+    # Head node
+    configure_hostname(node['ec2']['local_hostname'].split('.')[0].to_s)
+  end
+
+else
+  # Single Instance Type or master instance for Heterogeneous Instance Type
```

Comment:
master instance for HIT is not handled here

Reply:
Right, old comment